packages/big-design/src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,47 @@
+import { CloseIcon } from '@bigcommerce/big-design-icons';
+import React, { memo } from 'react';
+
+import { Text } from '../Typography';
+
+import { StyledChip, StyledCloseButton } from './styled';
+
+export interface ChipProps {
+  onDelete?(): void;
+}
+
+export const Chip: React.FC<ChipProps> = memo(({ children, onDelete }) => {
+  const label = typeof children === 'string' ? children : null;
+
+  const handleOnDelete = (event: React.SyntheticEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+
+    if (typeof onDelete === 'function') {
+      onDelete();
+    }
+  };
+
+  const renderDeleteButton = () =>
+    onDelete && (
+      <StyledCloseButton
+        variant="subtle"
+        onClick={handleOnDelete}
+        iconOnly={<CloseIcon size="medium" title="Delete" />}
+      ></StyledCloseButton>
+    );
+
+  return (
```

Comment:
Do we even want to render the chip if there is no `label`?
```suggestion
  return label ? (
```

Reply:
If we add the above, you should add another test with no children.

Reply:
I thought about that but I _think_ it's a bit easier to spot if we render a chip which looks wrong than to not render at all. Thoughts?

Reply:
Just from my own experience experimenting with `chips`, I think it's better if we display an empty chip than no chip at all. I was able to debug I wasn't sending the right text immediately.